jwst/extract_1d/extract.py

```diff
@@ -22,68 +26,67 @@
-# multi-integration data.
+# This is to prevent calling offset_from_offset multiple times for multi-integration data.
 OFFSET_NOT_ASSIGNED_YET = "not assigned yet"
 
 ANY = "ANY"
 """Wildcard for slit name.
 
 Extended summary
 ----------------
-For full-frame input data, keyword SLTNAME may not be populated, so the
-slit name will be set to this string to indicate that the first slit in
-the reference file should be used.
-A slit name in the reference file can also be ANY, in which case the
-reference information for that slit will be regarded as matching any slit
-name from the input data.
+For full-frame input data, keyword SLTNAME may not be populated, so the slit name will be set to this string to indicate 
+that the first slit in the reference file should be used.
+A slit name in the reference file can also be ANY, in which case the reference information for that slit will be 
+regarded as matching any slit name from the input data.
 """
 
 ANY_ORDER = 1000
 """Wildcard for spectral order number in a reference image.
 
 Extended summary
 ----------------
-If the reference file contains images, keyword SPORDER gives the order
-number of the spectrum that would be extracted using a given image in
-the reference file.
+If the reference file contains images, keyword SPORDER gives the order number of the spectrum that would be extracted 
+using a given image in the reference file.
 """
 
 HORIZONTAL = 1
 VERTICAL = 2
 """Dispersion direction, predominantly horizontal or vertical."""
 
-# This is intended to be larger than any possible distance (in pixels)
-# between the target and any point in the image; used by locn_from_wcs().
+# This is intended to be larger than any possible distance (in pixels) between the target and any point in the image;
+# used by locn_from_wcs().
 HUGE_DIST = 1.e20
 
 # These values are assigned in get_extract_parameters, using key "match".
-# If there was an aperture in the reference file for which the "id" key
-# matched, that's (at least) a partial match.  If "spectral_order" also
-# matched, that's an exact match.
-
+# If there was an aperture in the reference file for which the "id" key matched, that's (at least) a partial match.
+# If "spectral_order" also matched, that's an exact match.
 NO_MATCH = "no match"
 PARTIAL = "partial match"
 EXACT = "exact match"
 
-Aperture = namedtuple('Aperture', ['xstart', 'ystart', 'xstop', 'ystop'])
+
+class Aperture(NamedTuple):  # When python 3.6 is no longer supported, consider converting to DataClass
+    xstart: Union[int, float]
+    xstop: Union[int, float]
+    ystart: Union[int, float]
+    ystop: Union[int, float]
 
 
 class Extract1dError(Exception):
     pass
 
+
 class InvalidSpectralOrderNumberError(Extract1dError):
     """The spectral order number was invalid or off the detector."""
     pass
 
 
-def load_ref_file(refname):
+def load_ref_file(refname: str) -> dict:
     """Open the reference file.
```

Comment:
Is this used for both extract1d and apcorr ref files?

Reply:
No, it's not. In #4902 this function is renamed to specifically say that it's opening the extract1d file